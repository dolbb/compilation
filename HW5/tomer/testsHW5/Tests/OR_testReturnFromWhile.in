int addint(int x, int y){
	while(true){
		if(y==0)
			return x;
		x = x + 1;
		y = y - 1;
	}
}

byte addByte(byte x, int y){
	while(true){
		if(y==0)
			return x;
		x = x + 1b;
		y = y - 1;
	}
}

bool areEqual(int y, byte a)
{
	while(true){
		if(y==0)
			return a==0;
		y = y -1;
		a = a - 1b;
	}
}

void sayHello(int x)
{
	while(true)
	{
		if(x==0)
			return;
		print("hello\n");
		x = x -1;
	}
}

void main()
{
	int x = 0;
	x = addint(x,5);
	printi(x);
	print("\n");

	byte a = 0b;
	a = addByte(a,5);
	printi(a);
	print("\nThe numbers are ");
	if(not areEqual(x,a))
		print("not ");
	print("equal\n");
	sayHello(x);
}