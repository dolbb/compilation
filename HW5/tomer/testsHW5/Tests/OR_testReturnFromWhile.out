Loaded: ./exceptions.s
5
5
The numbers are equal
hello
hello
hello
hello
hello
